src/frontend/src/handler/create_table.rs

```diff
@@ -325,7 +325,14 @@ pub fn bind_sql_column_constraints(
     binder.bind_columns_to_context(table_name.clone(), column_catalogs)?;
 
     for column in columns {
-        let idx = binder.get_column_binding_index(table_name.clone(), &column.name.real_value())?;
+        let Some(idx) = column_catalogs
+            .iter()
+            .position(|c| c.name() == column.name.real_value())
+        else {
+            // It's possible that we don't follow the user defined columns in SQL but take the
+            // ones resolved from the source, thus missing some columns. Simply ignore them.
+            continue;
```

Comment:
It sounds ok, but I'm wondering what will happen before this PR?

Reply:
This only occurs when there are columns defined in SQL (as `ast::ColumnDef`) but not in the final schema (as `ColumnCatalog`). Therefore, this won't happen before this PR.

After this PR, we may encounter this when calling `REFRESH SCHEMA` on a purified definition with some columns to be removed. However, as we only care about generated columns here and generated columns will always be preserved, we can safely ignore them.